integrations/lambda/demo.ipynb

```diff
@@ -92,17475 +289,686 @@
     "\n",
-    "        records = []\n",
-    "        current_date = self.start_date\n",
-    "        max_measure_variance = 1.2\n",
-    "        while current_date <= self.end_date:\n",
-    "            # Each entity has a record for a single timestamp\n",
-    "            for entity in entities:\n",
-    "                dimensions = []\n",
-    "                for dimension_name in entity:\n",
-    "                    if dimension_name != \"most_recent_measures\":\n",
-    "                        dimension = {\n",
-    "                            \"Name\": dimension_name,\n",
-    "                            \"Value\": entity[dimension_name]\n",
-    "                        }\n",
-    "                        dimensions.append(dimension)\n",
+    "    def generate(self, start_date, end_date, reporting_frequency, num_entities, precision=\"MILLISECONDS\", generate_unique_options_fallback=False):\n",
+    "        num_currency_pairs = len(self._dimension_templates[0][\"unique_options\"])\n",
+    "        if num_entities > num_currency_pairs and not generate_unique_options_fallback:\n",
+    "            raise Exception(f\"num_entities ({num_entities}) was greater than the number of currency pairs ({num_currency_pairs})\")\n",
+    "        return super().generate(start_date, end_date, reporting_frequency, num_entities, generate_unique_options_fallback)\n",
     "\n",
-    "                # Set initial values for measures\n",
-    "                measures = {\n",
-    "                    \"energy_usage\": random.uniform(1, 10),\n",
-    "                    \"cpu_usage\": random.uniform(2, 20)\n",
-    "                } if current_date == self.start_date else {\n",
-    "                    \"energy_usage\": entity[\"most_recent_measures\"][\"energy_usage\"] + random.uniform(-max_measure_variance, max_measure_variance),\n",
-    "                    \"cpu_usage\": entity[\"most_recent_measures\"][\"cpu_usage\"] + random.uniform(-max_measure_variance, max_measure_variance)\n",
-    "                }\n",
-    "                entity[\"most_recent_measures\"] = measures\n",
     "\n",
-    "                record = {\n",
-    "                    \"Dimensions\": dimensions,\n",
-    "                    \"Time\": str(int(current_date.timestamp())), # Millisecond precision\n",
-    "                    \"Measures\": measures\n",
-    "                }\n",
-    "                records.append(record)\n",
-    "            current_date += self.frequency\n",
-    "        return records\n"
+    "class CustomDataGenerator(DataGenerator):\n",
+    "    def __init__(self, measure_templates: list, dimension_templates: list):\n",
+    "        self._measure_templates = measure_templates\n",
+    "        self._dimension_templates = dimension_templates"
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": 33,
-   "id": "ea01eea0",
+   "cell_type": "markdown",
+   "id": "17b8430c",
    "metadata": {},
-   "outputs": [
```

Comment:
If it helps with the review process I can reset the `dev-sample-notebook` branch so that this PR will essentially create the notebook. This deleted code is from outputs that exist in that branch.